### PR TITLE
02 run azimuth before kweight

### PR DESCRIPTION
**Pull request by maud-p**
#### Please link to the GitHub issue that this pull request addresses.
Copying the PR#706 https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/706 to:

•	roll back to what we had in https://github.com/AlexsLemonade/OpenScPCA-analysis/commit/f98da21d6f936791da10ca2d903e9661d9f409d0? (i.e., before all the k.weight debugging, etc.)
•	 use system() instead of source() for download-and-create-fetal-kidney-ref.R in 00_run_workflow.R

#703 

#### What is the goal of this pull request?

I write a RMarkdown script to 

1. download the data from the fetal kidney atlas
2. process it and create an azimuth compatible reference,

This will be used in the module to perform label transfer from the human fetal kidney atlas to the Wilms tumor samples.

#### Briefly describe the general approach you took to achieve this goal.

I used a different approach than described in the issue #703 .
I figured out that I can download the human fetal kidney data from cellxgene as a rds object.
url = "https://datasets.cellxgene.cziscience.com/40ebb8e4-1a25-4a33-b8ff-02d1156e4e9b.rds"

Like this, I didn't need to create a conda/renv enrironment. 
The dockerfile from this PR is the same as the dockerfile from the PR #704 .

Of note however, using your documentation, I have been able to build a conda/renv container that actually allow to run the scripts I did so far, in case we need it in the future :)

#### If known, do you anticipate filing additional pull requests to complete this analysis module?

Yes! The next one will be to implement the label transfer in the sample report. 

#### What is the name of your results bucket on S3?
here is the command I used to upload the result to the bucket:

```shell
scripts/sync-results.py cell-type-wilms-tumor-06 \
    --bucket researcher-008971640512-us-east-2 \
```

#### What types of results does your code produce (e.g., table, figure)?
The azimuth compatible reference in a format of 2 files:
- ref.Rds
- idx.annoy

When running the 00_fetal_reference_kidney.Rmd, these two files will be saved in the module folder/marker-genes. 

#### What is your summary of the results?
We build a reference that is compatible with the azimuth label transfer. 

### Provide directions for reviewers

I have one issue, using the RMarkdown 00_fetal_reference_kidney.Rmd, we can only build the reference manually running each chunk, but it do not work when we want to knittr as html report. 
I could isolate the problem to the AzimuthReference function, might be related to this issue: https://github.com/satijalab/azimuth/issues/219

#### What are the software and computational requirements needed to be able to run the code in this PR?

1. Run the docker container, 
2. open RStudio from http://localhost:8787/
3. in the module folder open the 00_fetal_reference_kidney.Rmd and run chunk by chunk (do not knittr!!)

### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._

#### Analysis module and review

- [x] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [x] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [x] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [x] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

### CODE REVIEW

**Review by jaclyn-taroni — COMMENTED** [review]
Hi @maud-p, this is looking really good! Thank you for being patient while we tried to iron out the issues with CI.

There are a few things I’d like to see before approving and merging:

1. Update the `results/README.md` file to reflect the current state of the results more accurately
2. Update various R Markdown files with the minor (docs, mostly) suggestions
3. Re-run all 40 samples and commit the HTML notebook output to the repository

After you make these changes, I think it is good to go. 

Removing the extraneous RDS files for samples 168-178 in your results bucket may also be helpful. You can check out the `--destructive-sync` option in `scripts/sync-results.py` by running (assuming you are in the repo root): `./scripts/sync-results.py --help`. 

As always, let us know if you have any questions, and thank you for your contributions!

File: analyses/cell-type-wilms-tumor-06/results/README.md [hunk 1]
```diff
@@ -3,3 +3,15 @@
...
+# Azimuth compatible fetal kidney reference
+
+To perform label transfer using Azimuth and the fetal kidney atlas, a reference is build (in 00_fetal_reference_kidney.Rmd) using the fetal_full.Rds object download from :
+"https://datasets.cellxgene.cziscience.com/40ebb8e4-1a25-4a33-b8ff-02d1156e4e9b.rds"
+
+The result of 00_fetal_reference_kidney.Rmd can be found in the S3 bucket name `researcher-008971640512-us-east-2` named ref.Rds and idx.annoy
+
+
+# Clustering and label transfer from fetal references
+
```
[the remark this answers is not in the document]
Comment (maud-p) [reply analyses/cell-type-wilms-tumor-06/results/README.md:17]: I think it do not matter, so I'll just do it now :)

**Comment by maud-p** [comment]
Thank you very much @jaclyn-taroni ! I'll do the last changes on the `Read.me` file right now and run the analysis over the evening/night. If everything fine, I should commit the 40 reports by tomorrow morning :) (European time, Vienna).

Thank you so much, looking forward the next steps :D

**Comment by maud-p** [comment]
@jaclyn-taroni FYI, the extraneous RDS files for samples 168-178 in my results bucket should have been removed using --destructive-sync !

**Review by jaclyn-taroni — APPROVED** [review]
Thanks for the updates, @maud-p! I am going to approve at this time 👍🏻  

There are a couple things that will make future PRs easier until we figure out how to approach testing the Azimuth notebooks. I will make those changes here and then get this merged in! 🥳

**Comment by jaclyn-taroni** [comment]
https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/737/commits/5afefe3e96f8e06a49c839f2d9729af93aa1c7e2 is just the HTML output of the notebooks after the change in https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/737/commits/7e877edce5323c5c7247a9088712c13ac577868f, which passed CI: https://github.com/AlexsLemonade/OpenScPCA-analysis/actions/runs/10656117562/job/29534402700

Merging without the check on running the module.

**Comment by maud-p** [comment]
Thank you so much @jaclyn-taroni ! It is great to see the merge in the main branch :) I'll create a new branch from here and continue with PR #704 ! Thanks!!

**Pull request merged**